File: .NET-app/MorkosiaPrepaLeague/Models/Core/AttendanceRequest.cs
using System;

namespace MorkosiaPrepaLeague.Models.Core
{
    public class AttendanceRequest : CosmosDocumentBase
    {
        public string DocumentType { get; } = "AttendanceRequest";

        public string LeagueId { get; set; } = string.Empty;

        public string TransferWindowId { get; set; } = string.Empty;

        public string ActivityId { get; set; } = string.Empty;

        public string StudentUserId { get; set; } = string.Empty;

        public string StudentName { get; set; } = string.Empty;

        public string ActivityName { get; set; } = string.Empty;

        public DateTime ActivityDate { get; set; }

        public AttendanceRequestStatus Status { get; set; } = AttendanceRequestStatus.Pending;

        public DateTime RequestedAt { get; set; } = DateTime.UtcNow;

        public DateTime? ProcessedAt { get; set; }

        public string? ProcessedByUserId { get; set; }
    }

    public enum AttendanceRequestStatus
    {
        Pending,
        Approved,
        Rejected
    }
}

File: src/app/models/attendance_request.py
from datetime import datetime, timezone
from enum import IntEnum
from typing import Optional

from beanie import Indexed
from pydantic import Field

from app.models.base import DocumentBase


class AttendanceRequestStatus(IntEnum):
    PENDING = 0
    APPROVED = 1
    REJECTED = 2


class AttendanceRequest(DocumentBase):
    document_type: str = "AttendanceRequest"
    league_id: Indexed(str) = ""  # type: ignore[valid-type]
    transfer_window_id: str = ""
    activity_id: str = ""
    student_user_id: str = ""
    student_name: str = ""
    activity_name: str = ""
    activity_date: datetime
    status: AttendanceRequestStatus = AttendanceRequestStatus.PENDING
    requested_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    processed_at: Optional[datetime] = None
    processed_by_user_id: Optional[str] = None

    class Settings:
        name = "attendance_requests"
